examples/github/build-and-deploy-mender-artifact.yml

```diff
@@ -0,0 +1,127 @@
+  MENDER_ARTIFACTS_CI_ARTIFACT_NAME: mender-artifacts # GitHub Actions artifact name
+  MENDER_ARTIFACTS_CI_ARTIFACT_PATH: mender-artifacts # GitHub Actions artifact path
+
+jobs:
+  #
+  # Build Mender artifact
+  #
+  build:
+    runs-on: ubuntu-latest
+    # conrainer with pre-installed mender-artifact and mender-cli tools
+    container:
+      image: mendersoftware/mender-ci-tools:master
+    outputs:
+      mender_release_name: ${{ steps.release_name.outputs.mender_release_name }}
+      mender_devices_group: ${{ steps.deployment_group.outputs.mender_devices_group }}
+      mender_deployment_name: ${{ steps.deployment_name.outputs.mender_deployment_name }}
+      mender_artifact_name: ${{ steps.create_artifact.outputs.mender_artifact_name }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Generrate Mender Release Name
+        id: release_name
+        run: |
+          [ ! -d ${MENDER_ARTIFACTS_CI_ARTIFACT_PATH} ] && mkdir -p ${MENDER_ARTIFACTS_CI_ARTIFACT_PATH}
+          APP_NAME=change-the-world-to-better
+          VERSION="$(($RANDOM%9+1)).$(($RANDOM%9+1)).$(($RANDOM%9+1))"
+          MENDER_RELEASE_NAME=${APP_NAME}_v${VERSION}
+          echo "MENDER_RELEASE_NAME=${MENDER_RELEASE_NAME}" >> ${GITHUB_ENV}
+          echo "mender_release_name=${MENDER_RELEASE_NAME}" >> ${GITHUB_OUTPUT}
+      - name: Choose Mender Deployment Group
+        id: deployment_group
+        run: |
+          if [[ "${GITHUB_BASE_REF}" == "production" ]]; then
+            MENDER_DEPLOYMENT_DEVICES_GROUP=${MENDER_DEPLOYMENT_DEVICES_GROUP_PROD}
+          elif [[ "${GITHUB_BASE_REF}" == "main" ]]; then
+            MENDER_DEPLOYMENT_DEVICES_GROUP=${MENDER_DEPLOYMENT_DEVICES_GROUP_DEV}
+          else
+            echo "ERROR: workflow execution on not supported branch"
+            exit 1
+          fi
+          echo "mender_devices_group=${MENDER_DEPLOYMENT_DEVICES_GROUP}" >> ${GITHUB_OUTPUT}
+      - name: Generate Mender Deployment Name
+        id: deployment_name
+        run: |
+          COMMIT_SHORT_SHA="${GITHUB_SHA:0:8}"
+          MENDER_DEPLOYMENT_NAME=${MENDER_RELEASE_NAME}_${COMMIT_SHORT_SHA}
+          echo "mender_deployment_name=${MENDER_DEPLOYMENT_NAME}" >> ${GITHUB_OUTPUT}
+      - name: Create Mender Artifact
+        id: create_artifact
+        run: |
+          MENDER_ARTIFACT_NAME=artifact_$(( $RANDOM % 9999 + 1000 )).mender
+          echo /home/root/.ssh > dest_dir
+          echo authorized_keys > filename
+          echo key > authorized_keys
+          mender-artifact write module-image \
+            -T single-file \
+            --device-type raspberrypi4 \
+            -o ${MENDER_ARTIFACTS_CI_ARTIFACT_PATH}/${MENDER_ARTIFACT_NAME} \
+            -n ${MENDER_RELEASE_NAME} \
+            --software-name authorized_keys \
+            --software-version 1.0 \
+            -f dest_dir \
+            -f filename \
+            -f authorized_keys
```

Comment:
@lluiscampos It's the example taken from our [docs](https://docs.mender.io/artifact-creation/create-an-artifact#create-an-application-update-artifact). I didn't really get your intention here. Can you please give some more context?

Reply:
> This won't really work. `single-file` update module accepts one file. 

Yes, sorry. I got this wrong. Sorry for the confusion. That example is incorrect, though, but I will follow-up on that independently.

> We should install one Python script assuming it is defined somewhere else in a previous step

We need to do something more meaningful than this. My suggestion was a Python script but now I see that to really make it easy we need the generator.

So, let's leave this as-is and I am creating a follow-up task in the epic to get the generator and install something useful. This will apply to all examples.